### PR TITLE
Fix #115: Adjust HTTP response code for invalid CSRF token

### LINKED ISSUE #115: Inappropriate HTTP 400 for CSRF token errors
Yii currently responds to requests that lack a valid CSRF token with a `BadRequestHTTPExeption`, which translates to HTTP error `400 Bad Request`:

https://github.com/yiisoft/yii2/blob/master/framework/web/Controller.php#L166

I suggest that this error code is semantically inappropriate. The relevant section of [RFC 7231](https://tools.ietf.org/html/rfc7231#section-6.5.1) specifies that `400 Bad Request` is intended for:
> client error (e.g., malformed request syntax, invalid request message framing, or deceptive request routing)

I understand this to mean that `400 Bad Request` indicates something is wrong with the HTTP request itself rather than its payload, and it seems to me that CSRF validation failure does not constitute an error in the HTTP protocol; CSRF tokens aren't part of HTTP, and an HTTP request without a valid CSRF token is still perfectly valid as far as HTTP is concerned.

More appropriate responses might be:
* `422 Unprocessable entity` ([used by Rails](https://github.com/rails/rails/blob/fdd5894aa985651fb99e0ef1c2d3f3f604b2f659/actionpack/lib/action_dispatch/middleware/exception_wrapper.rb#L17))
* `403 Forbidden`

[This discussion on Software Engineering Stack Exchange ](https://softwareengineering.stackexchange.com/q/329229) may also be of interest.

### PR DESCRIPTION
**Pull request by samdark**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ✔️
| Tests pass?   | ✔️
| Fixed issues  | #115

### CODE REVIEW

File: tests/Middleware/CsrfTest.php [hunk 1]
```diff
@@ -79,31 +79,31 @@ public function getIsAlwaysAllowed()
...
     /**
      * @test
      */
-    public function emptyTokenInSessionResultIn400()
+    public function emptyTokenInSessionResultIn403()
     {
         $middleware = $this->createCsrfMiddlewareWithToken('');
         $response = $middleware->process($this->createPostServerRequestWithBodyToken($this->generateToken()), $this->createRequestHandler());
-        $this->assertEquals(400, $response->getStatusCode());
+        $this->assertEquals(403, $response->getStatusCode());
```
Comment (roxblnfk) [inline tests/Middleware/CsrfTest.php]: Should we check that with an invalid token, the `_csrf` key is deleted from the session?

**Comment by schmunk42** [comment]
I'd rather tend to error code `422`, see also https://github.com/yiisoft/yii-web/issues/115#issue-495235305

A `403` is more tied to a wrong login or auth-token, IMHO.

File: tests/Middleware/CsrfTest.php (hunk 1 above)
Comment (samdark) [reply tests/Middleware/CsrfTest.php]: I think that would be too much for this test.

**Review by roxblnfk — APPROVED** [review]

**Comment by samdark** [comment]
Thanks for review, everyone.

**Pull request merged**